SSTOOLKIT-67 (#47)
nordic-institute/X-Road-Security-Server-toolkit: docs/xroad_security_server_toolkit_user_guide.md

The X-Road Security Server Toolkit package can be installed using PIP:

```
$ pip install --index-url http://xroad-toolkit.s3-website-eu-west-1.amazonaws.com/xrdsst xrdsst
```

The downloaded and signed packages can be verified with a public key:

> **The public key can be obtained from : https://pgp.mit.edu/ by typing ``jenkins@niis.org`` into the search string field.**

Reviewer: Here it would be good to list key fingerprint explicitly, like in other X-Road package documentations. The import command then can also be given together with ``--with-fingerprint`` option, -- this search box at pgp.mit.edu has been verified to work ATM, but is not very graceful and it would be easily possible to add spam keys that also show up when searching for 'jenkins@niis.org'.

Is this key not available at SKS keyserver pools?

Author: This key has been added according to some instructions: 
Upload to sks keyserver pool
Select a keyserver from the pool that provides a webinterface, for instance https://pgp.mit.edu
Paste the content of the file "john_doe.pub" unmodified and including the ----BEGIN and ---END lines into the form
Click on "submit this key to the keyserver!"

So I do not see any issue with that.

I can add the fingerprint to UG.

Author: Added

Reviewer: Now please just change import command to also verify fingerprint and this is good to merge and simple to use.

Author: Changed

The X-Road Security Server Toolkit package can be installed using PIP:

```
$ pip install --index-url http://xroad-toolkit.s3-website-eu-west-1.amazonaws.com/xrdsst xrdsst
```

The packaged have been signed with a private GPG key:

> ```

Reviewer: The signing part is not needed for the UG.

Author: Removed

$ gpg --detach-sign -a xrdsst-*.tar.gz
$ gpg --detach-sign -a xrdsst-*.whl
```

The downloaded and signed packages can be verified with a public key:
```

> $ gpg --import key.gpg

Reviewer: I do not see instructions about where to acquire the ``key.gpg``, please add them.

Author: Updated

The X-Road Security Server Toolkit package can be installed using PIP:

```

> $ pip install -i http://xroad-toolkit.s3-website-eu-west-1.amazonaws.com/ xrdsst --trusted-host xroad-toolkit.s3-website-eu-west-1.amazonaws.com

Reviewer: Does it work with ``pip``, it seems it needs to be ``pip3`` always, no?

Author: Commented in the UG, that pip or pip3 should be used, whichever is used. For example locally I use "pip3", but on Jenkins it is "pip", it depends on symbolic links, like "python3" could also be run as "python". If I would put only "pip3" there, somebody may ask, why not "pip".

```

> ** The public key ``jenkins@niis.org`` has been added to an SKS server: pool.sks-keyservers.net **

Reviewer: ```suggestion
Package signing public key for can be retrieved from SKS keyserver pool (pool.sks-keyservers.net), key fingerprint is ``BEC35825BBAB4288933F0354116AC90A8F670D74``, publisher ``Jenkins (X-Road Development Signing Key) <jenkins@niis.org>``.
```

Author: Amended

The fingerprint of that key:
```
pub   rsa4096 2018-05-31 [SC]
      BEC35825BBAB4288933F0354116AC90A8F670D74
uid           [ unknown] Jenkins (X-Road Development Signing Key) <jenkins@niis.org>

> ```

Reviewer: If you accept the suggestion at line 108, the lines 110..115 could be removed.

Author: Accepted the suggestion and removed the other lines.

> The downloaded and signed packages can be verified with the public key:

Reviewer: ```suggestion
Downloaded packages with detached signatures can be verified after adding signing public key to local keyring:
```

Author: Added these changes

```

> $ gpg --keyserver pool.sks-keyservers.net --search-keys  BEC35825BBAB4288933F0354116AC90A8F670D74

Reviewer: ```suggestion
# Retrieve the key, accept into keyring. 
$ gpg --keyserver pool.sks-keyservers.net --search-keys BEC35825BBAB4288933F0354116AC90A8F670D74
```

> $ gpg --verify xrdsst-*.tar.gz.asc xrdsst-*.tar.gz 

Reviewer: ```suggestion
$ gpg --verify xrdsst-*.tar.gz.asc
```

^^ 
There shoud be no need to add ``tar.gz`` files separately, if their file prefix matches the signature file prefix. In fact, I think the wildcards might not work when you have many of these packages at the same folder. Did you try this command before adding it here? With single undetached signature it simply gives:

```
$ ls
xrdsst-0.2.2a0.tar.gz  xrdsst-0.2.2a0.tar.gz.asc
# Lo and behold
$ gpg --verify xrdsst-*.tar.gz.asc xrdsst-*.tar.gz
gpg: not a detached signature
# Without specific .tar.gz it fares better
$ gpg --verify xrdsst-*.tar.gz.asc 
gpg: Signature made Thu 11 Feb 2021 18:04:09 EET
gpg:                using RSA key 116AC90A8F670D74
gpg: Good signature from "Jenkins (X-Road Development Signing Key) <jenkins@niis.org>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: BEC3 5825 BBAB 4288 933F  0354 116A C90A 8F67 0D74
gpg: WARNING: not a detached signature; file 'xrdsst-0.2.2a0.tar.gz' was NOT verified!
```

Author: Hard-coded signature file name, so no need for wildcards anymore, they were there because of the version number, which can not be hard-coded, but now the signature file is renamed to have always the same name.
